packages/datatrak-web/src/features/ProjectSelectForm.tsx

```diff
@@ -31,99 +31,100 @@ const ListWrapper = styled.div`
-    Item
-  ) : (
-    <Tooltip title="Request project access" arrow>
-      {Item}
-    </Tooltip>
-  );
-};
-
 interface ProjectSelectFormProps {
   projectId?: Project['id'];
   variant?: 'modal' | 'page';
   onClose: () => void;
+  onRequestAccess?: (projectCode: Project['code']) => void;
 }
 
 export const ProjectSelectForm = ({
   projectId,
   onClose,
   variant = 'page',
+  onRequestAccess,
 }: ProjectSelectFormProps) => {
   const navigate = useNavigate();
   const [selectedProjectId, setSelectedProjectId] = useState(projectId);
   const { data: projects, isLoading } = useProjects();
-  const { mutate, isLoading: isConfirming } = useEditUser(onClose);
+  const { mutate, isLoading: isConfirming, error } = useEditUser(onClose);
   const onConfirm = () => {
     mutate({ projectId: selectedProjectId! });
   };
 
-  const onRequestAccess = () => {
-    navigate(ROUTES.REQUEST_ACCESS);
-    if (variant === 'modal') {
-      onClose();
+  const handleRequestAccess = project => {
+    if (variant === 'modal' && onRequestAccess) {
+      onRequestAccess(project.code);
+    } else {
+      navigate({
+        pathname: ROUTES.REQUEST_ACCESS,
+        search: `?project=${project?.code}`,
+      });
     }
   };
 
   const onSelect = project => {
     if (project.hasAccess) {
       setSelectedProjectId(project.value);
     } else {
-      onRequestAccess();
+      handleRequestAccess(project);
     }
   };
 
-  const projectOptions = projects?.map(({ entityName, id }, index) => ({
-    name: entityName,
-    value: id,
-    selected: id === selectedProjectId,
-    // Todo: get hasAccess from api
-    hasAccess: index < 6,
-  }));
+  const getProjectIcon = (hasAccess: boolean, hasPendingAccess: boolean) => {
+    if (hasPendingAccess) return <WatchLater />;
+    if (!hasAccess) return <Lock />;
+    return null;
+  };
+
+  const getProjectTooltip = (hasAccess: boolean, hasPendingAccess: boolean) => {
+    if (!hasAccess && !hasPendingAccess) return 'Request project access';
+    return '';
+  };
+
+  const getProjectContent = (name: string, hasPendingAccess: boolean) => {
+    // customise the tooltip to appear over the project name if there is pending access, instead of the default of over the whole list item
+    if (hasPendingAccess)
+      return (
+        <Tooltip title="Approval in progress">
+          <span>{name}</span>
+        </Tooltip>
+      );
+    return name;
+  };
+
+  const getFormattedProjects = () => {
+    return projects?.map(({ name, code, hasAccess, id, hasPendingAccess }) => {
+      const icon = getProjectIcon(hasAccess, hasPendingAccess);
+      const tooltip = getProjectTooltip(hasAccess, hasPendingAccess);
+      const content = getProjectContent(name, hasPendingAccess);
+      return {
+        content,
+        value: id,
+        code,
+        selected: id === selectedProjectId,
+        icon,
+        tooltip,
+        button: !hasPendingAccess,
+        appearsDisabled: hasAccess === false,
+        hasAccess
```

Comment:
I find that this approach can get unwieldy as we end up with a prop to support every style difference and the list of props can get quite long if there are lots of different implementations of the components. It looks like there are only 3 use cases of the List in the design so this list of props should be the full extent so it should be all good and I do like how this approach is more convenient to use once the component is set up.